### PR TITLE
Refs #29784 -- Switched to https:// links where available.

### PR DESCRIPTION
**Pull request by jdufresne**
Changed Python f-string link to use intersphinx.

Normalized docs.python.org URLs to omit the version.

https://code.djangoproject.com/ticket/29784

### CODE REVIEW

**Comment by jdufresne** [comment]
I've done this same change on a number of different repositories now. I automated this change using the script:

https://github.com/jdufresne/commit-messages/blob/master/scripts/checkhttp.py

At a high level, this script does the following:

1. Scans all text files for strings that look like a URL starting with `http://`
2. Tries fetching the same URL but with `https://`
3. Check the final destination is a 200 response with protocol `https://` (to avoid redirects back to `http://`)
4. If it looks good, replace all occurrences

After running the script, I analyze the results locally using a visual diff tool (I like [meld](https://wiki.gnome.org/Apps/Meld)). I undo the changes for tests that look to be intentionally using an `http://` URL (Django has a lot of these). I also try to remove `en/latest/` paths from readthedocs that didn't previously exist. I manually test a bunch of URLs especially any that look suspicious with a wildly different path. I re-analyze the results using GitHub's diff tool (I find looking at a diff in multiple ways can help emphasize different differences).

There are classes of URLs we can immediately be confident with:


* https://docs.djangoproject.com
* https://code.djangoproject.com
* https://docs.python.org
* https://bugs.python.org
* https://blog.python.org
* https://github.com
* https://www.sqlite.org
* https://stackoverflow.com

Perhaps some others too.

So, in summary, a script has confirmed that all these changes result in `https://` URLs that 200 and several have been manually tested.

Please feel free to re-run the script to verify or do additional manual testing.

**Comment by carltongibson** [comment]
Thanks @jdufresne. Good explanation. Sounds fine. (I will give it a run and review.) 

@IgnatiusFlo I mainly CC-d you here so you knew what was coming and didn't repeat effort. If you focus on the remainder of #29784 when you have time, that would be super. Thanks!

**Comment by jdufresne** [comment]
> Can the last commit message start with 'Fixed'?

This PR only fixes one aspect of the ticket. The ticket is asking for a full audit of the external links on the settings docs. Links using https:// is just one aspect of this audit. From the ticket, see the text:

> Some, like ​https://www.owasp.org/index.php/HTTPOnly should be replaced with a link to more up to date information.

This PR isn't addressing that concern. So I think changing this to "Fixed" would close the ticket prematurely and skip over some intended changes desired by the ticket.

I decided to handle https:// links separately because it is a global change, not localized to just one page of the docs.

---

Rebased.

**Comment by claudep** [comment]
Of course, sorry for my poor comment.
However, I'm not sure about the feasability of a global audit. It may be a rather huge task, and wonder if we should ask the reporter to open tickets for the links he noticed. It's a bit like "Audit the documentation for grammar constructs...".

**Comment by carltongibson** [comment]
My thought was the 80:20 would be feasible, and good enough. (Plus, I think @IgnatiusFlo volunteered, so it should t hang too log.)

File: django/contrib/admin/static/admin/img/README.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 All icons are taken from Font Awesome (http://fontawesome.io/) project.
 The Font Awesome font is licensed under the SIL OFL 1.1:
-- http://scripts.sil.org/OFL
+- https://scripts.sil.org/cms/scripts/page.php?site_id=nrsi&id=OFL
```
Comment (claudep) [inline django/contrib/admin/static/admin/img/README.txt]: Using https is fine, but I would keep the shorter URL even if it redirects. The longer URL seems less robust.

File: django/template/defaulttags.py [hunk 2]
```diff
@@ -1131,7 +1131,7 @@ def now(parser, token):
     """
     Display the date, formatted according to the given string.
 
-    Use the same format as PHP's ``date()`` function; see http://php.net/date
+    Use the same format as PHP's ``date()`` function; see https://secure.php.net/manual/en/function.date.php
```
Comment (claudep) [inline django/template/defaulttags.py]: Here I would also limit the change to https.
